lib/openapi_parser/concerns/findable.rb

```diff
@@ -46,8 +46,8 @@ def purge_object_cache
   private
 
     def find_remote_object(reference)
-      reference_uri = URI(reference)
-      fragment = reference_uri.fragment
+      uri, fragment = reference.split("#")
```

Comment:
When the reference include `#` twice, this code doesn't work well.

JSON pointer use all character without `/` and `~` so it can have `#`.
```
 unescaped       = %x00-2E / %x30-7D / %x7F-10FFFF
         ; %x2F ('/') and %x7E ('~') are excluded from 'unescaped'
```
https://datatracker.ietf.org/doc/html/rfc6901

So when the JSON Pointer have `#`, the reference have two or more `#` .

Reply:
Thank you for reviewing.

I fix this corner case and add  some test case.

Please confirm.

Reply:
Your code is correct, sorry I misunderstanding.

- [Reference Object](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#referenceObject
) have [JSON Reference](https://tools.ietf.org/html/draft-pbryan-zyp-json-ref-03).
- JSON Reference have [URI](https://datatracker.ietf.org/doc/html/rfc3986) and that's fragment is [JSON pointer](https://datatracker.ietf.org/doc/html/rfc6901)

JSON Pointer can be included any character, but this JSON Pointer must be valid URI fragment.
URI fragment doesn't include `#` so this JSON Pointer must not have `#`.

This means my comment is incorrect.. sorry....
Please revert 2a6979e and 5d74a3d (or force push) 🙇 🙇 🙇 🙇 🙇

Reply:
Thank you for your reply!!

This fix is fine, don't need to revert.

I would like to merge it as it is.